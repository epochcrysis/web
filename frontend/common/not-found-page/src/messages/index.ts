import { defineMessages } from 'react-intl'

export const namespace = '@client-starter/not-found-page'

export default defineMessages({
  pageNotFound: {
    id: `${namespace}.page_not_found`,
    defaultMessage: 'Страница не найдена',
  },
})
