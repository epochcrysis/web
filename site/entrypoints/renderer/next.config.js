const withPlugins = require('next-compose-plugins')
const { withWorkspaces } = require('@atls/next-config-with-pnp-workspaces')

const nextConfig = {
  experimental: {
    esmExternals: 'loose',
  },
}

module.exports = withPlugins([withWorkspaces], nextConfig)
