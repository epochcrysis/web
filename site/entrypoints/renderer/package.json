{
  "name": "@site/renderer-entrypoint",
  "version": "0.0.0",
  "license": "MIT",
  "files": [
    "dist/",
    "next.config.js"
  ],
  "scripts": {
    "build": "yarn next build src && cp -r src/.next dist/.next",
    "dev": "yarn next dev src",
    "start": "node dist/index.js"
  },
  "dependencies": {
    "@apollo/client": "^3.5.7",
    "@atls/next-config-with-pnp-workspaces": "^0.0.2",
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "@site/index-page": "workspace:0.0.0",
    "@ui/theme": "workspace:0.0.0",
    "express": "^4.17.1",
    "next": "^11.1.0",
    "next-compose-plugins": "2.2.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@types/express": "^4.17.12",
    "@types/node": "^16.11.10",
    "@types/react": "^17.0.37",
    "@types/react-dom": "^17.0.11",
    "typescript": "^4.4.2"
  },
  "externalDependencies": {
    "next/babel": "*",
    "next": "*"
  }
}
