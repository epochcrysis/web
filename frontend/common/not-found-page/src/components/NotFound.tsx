import React from 'react'
import { injectIntl } from 'react-intl'
import messages from '../messages'

const NotFound = ({ intl }) => (
  <div>{intl.formatMessage(messages.pageNotFound)}</div>
)

export default injectIntl(NotFound)
