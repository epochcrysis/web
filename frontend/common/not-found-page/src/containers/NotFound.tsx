import React from 'react'
import { connect } from 'react-redux'
import { Route } from 'react-router-dom'
import NotFound from '../components/NotFound'

export default connect()(props => (
  <Route
    render={({ staticContext }) => {
      if (staticContext) {
        staticContext.statusCode = 404 // eslint-disable-line no-param-reassign
      }

      return <NotFound {...props} />
    }}
  />
))
