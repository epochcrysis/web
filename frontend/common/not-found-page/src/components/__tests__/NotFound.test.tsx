import React from 'react'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import messages from '../../messages'
import NotFound from '../NotFound'

describe('NotFound page', () => {
  let intl

  beforeEach(() => {
    const intlProvider = new IntlProvider(
      {
        defaultLocale: 'ru',
        locale: 'ru',
        messages,
      },
      {}
    )

    intl = intlProvider.getChildContext().intl // eslint-disable-line prefer-destructuring
  })

  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const notFound = renderer
        .create(<NotFound.WrappedComponent intl={intl} />)
        .toJSON()

      expect(notFound).toMatchSnapshot()
    })
  })
})
