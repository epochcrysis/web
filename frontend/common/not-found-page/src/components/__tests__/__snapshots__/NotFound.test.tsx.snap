// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page snapshots should match latest render snapshot 1`] = `
<div>
  Страница не найдена
</div>
`;
